From 59f8403616e8db6dcf8f3489c44b61524044fe64 Mon Sep 17 00:00:00 2001
From: Fernando Rodrigues <alpha@sigmasquadron.net>
Date: Thu, 2 Jan 2025 04:28:42 +0000
Subject: [PATCH] Function inclusion fixes for GCC 14

This patch includes <algorithm> in bsa_file.cpp and <list> in
charactermanager.hpp to prevent a build failure with GCC 14.

Signed-off-by: Fernando Rodrigues <alpha@sigmasquadron.net>
---
 apps/openmw/mwstate/charactermanager.hpp | 1 +
 components/bsa/bsa_file.cpp              | 1 +
 2 files changed, 2 insertions(+)

diff --git a/apps/openmw/mwstate/charactermanager.hpp b/apps/openmw/mwstate/charactermanager.hpp
index 8b3f2b8f8f..fac73b3d44 100644
--- a/apps/openmw/mwstate/charactermanager.hpp
+++ b/apps/openmw/mwstate/charactermanager.hpp
@@ -4,6 +4,7 @@
 #include <boost/filesystem/path.hpp>

 #include "character.hpp"
+#include <list>

 namespace MWState
 {
diff --git a/components/bsa/bsa_file.cpp b/components/bsa/bsa_file.cpp
index 4f795ec0d4..38e97b267b 100644
--- a/components/bsa/bsa_file.cpp
+++ b/components/bsa/bsa_file.cpp
@@ -25,6 +25,7 @@

 #include <components/files/constrainedfilestream.hpp>

+#include <algorithm>
 #include <cassert>

 #include <boost/filesystem/path.hpp>
--
2.47.0

